[Settings] New UX

niels9001 opened the pull request:
## Summary of the Pull Request
This PR introduces a new look and feel for Settings. Advantages:

- New, beautiful look and feel :)
- Easier to maintain: new standardized controls can now be edited once and will trickle down to all pages!
- Now inline with Windows 11 Settings UX
- Many accessibility improvements

![NewSettings](https://user-images.githubusercontent.com/9866362/129934110-c6dcbc05-af2c-44d4-a309-9d65fb1a7470.gif)




Issues resolved:

#12507 - Chosing a mode is now a ComboBox vs. RadioButtons
#12028 - Links are now more descriptive ("See what's new in this version" vs. "Version 0.43" or "Read more")
#12023 - Updated link name "Learn more about Color Picker" vs. "Learn more"
#12011 - Narrator now announces ToggleSwitch
#11988 - Correct themeresources are now used
#11860 - Now uses the same look and feel as W11 Settings
#10801 - "Direct activation command" is now used
#10635 - Is now keyboard accessible with dropdown buttons
#9412 - Now using the new visual styles
#9345 - Setting header is now aligned
#7058 - ToolTip is no longer used - now a text description that is always visible
#11196 - HotkeyControls are now disabled whenever the module is disabled
#10725 - Hyperlinks replaced with HyperLinkButtons
#12774 - Colored icons are now used
#9176 - No longer making use of this icon
#10778  - VCM is redesigned as well
#12832 - Changed label to "Restart PowerToys as administrator"
#12632 - Changed label from "Off (Passive)" to "Inactive"
#10038 - We now use an expander, so that resolves this issue.
#9987 - OOBE now launches with the same theme applied as Settings
#11366 - New update-ing UI
#6595 - There's now a release notes button next to the current version number

## Quality Checklist

- [X] **Linked issue:** #12507 #12028 #12023 #12011 #11988 #11860 #10801 #10635 #9412 #9345 #7058 #11196 #12774 #10725 #10778 #9176 #12832 #12632 #10038 #9987 #11366 #6595
- [ ] **Communication:** I've discussed this with core contributors in the issue. 
- [ ] **Tests:** Added/updated and all pass
- [ ] **Installer:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Docs:** Added/ updated
- [ ] **Binaries:** Any new files are added to WXS / YML
   - [ ] No new binaries
   - [ ] [YML for signing](https://github.com/microsoft/PowerToys/blob/master/.pipelines/pipeline.user.windows.yml#L68) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/master/installer/PowerToysSetup/Product.wxs) for new binaries

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.

Commit:
Removed hardcoded background

Commit:
Background

Commit:
Updated pages

Commit:
Merge branch 'master' into users/niels9001/settings-newstyle

Commit:
Added OOBE to shellpage

Commit:
Margin fixes

Commit:
Fix

Commit:
Resource update

Commit:
Resource string update

Commit:
Removed reference from installer file

htcfreek commented:
@niels9001 
- Checks are failing.
- I think the "Welcome to PT" button/link should be white and not black in dark theme. Or is this default on WinUI 2.6 style? Why is it not the same on ligth and dark mode?
- I am not sure. But it seems for me that the space between symbol and text is to big on the "Welcome to PT" button/link.

jaimecbernardo commented:
Hi @niels9001 ,
This is pretty weird, but the update to WINUI 2.6 is causing some weird errors on my machine.
I can open Settings when running from Visual Studio, but when I try running from the build folder directly or from an installed version, settings doesn't open.
I've got these 2 errors in the Event Viewer each time I try to open Settings:
```
Application: PowerToys.Settings.exe
CoreCLR Version: 4.700.21.26205
.NET Core Version: 3.1.16
Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FFB18A41EE7 (00007FFB188B0000) with exit code c0000005.
```
and
```
Faulting application name: PowerToys.Settings.exe, version: 0.0.1.0, time stamp: 0x609c77fe
Faulting module name: coreclr.dll, version: 4.700.21.26205, time stamp: 0x609c312a
Exception code: 0xc0000005
Fault offset: 0x0000000000191ee7
Faulting process id: 0x420c
Faulting application start time: 0x01d77f0d9e095298
Faulting application path: D:\prog\janeasystems\PowerToys\x64\Release\Settings\PowerToys.Settings.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.16\coreclr.dll
Report Id: f99ac88b-1d90-4e6c-8c50-440fcf27980f
Faulting package full name: 
Faulting package-relative application ID: 
```

I've tried to understand which commit started causing this and it seems to have been: https://github.com/microsoft/PowerToys/commit/1bd702340bb56ecedb198e160a91cce0ccd61b06

I still haven't installed the "Windows 10, version 20H2", in case that's relevant.

davidegiacometti commented:
> Hi @niels9001 ,
> 
> This is pretty weird, but the update to WINUI 2.6 is causing some weird errors on my machine.
> 
> I can open Settings when running from Visual Studio, but when I try running from the build folder directly or from an installed version, settings doesn't open.
> 
> I've got these 2 errors in the Event Viewer each time I try to open Settings:
> 
> ```
> 
> Application: PowerToys.Settings.exe
> 
> CoreCLR Version: 4.700.21.26205
> 
> .NET Core Version: 3.1.16
> 
> Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FFB18A41EE7 (00007FFB188B0000) with exit code c0000005.
> 
> ```
> 
> and
> 
> ```
> 
> Faulting application name: PowerToys.Settings.exe, version: 0.0.1.0, time stamp: 0x609c77fe
> 
> Faulting module name: coreclr.dll, version: 4.700.21.26205, time stamp: 0x609c312a
> 
> Exception code: 0xc0000005
> 
> Fault offset: 0x0000000000191ee7
> 
> Faulting process id: 0x420c
> 
> Faulting application start time: 0x01d77f0d9e095298
> 
> Faulting application path: D:\prog\janeasystems\PowerToys\x64\Release\Settings\PowerToys.Settings.exe
> 
> Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.16\coreclr.dll
> 
> Report Id: f99ac88b-1d90-4e6c-8c50-440fcf27980f
> 
> Faulting package full name: 
> 
> Faulting package-relative application ID: 
> 
> ```
> 
> 
> 
> I've tried to understand which commit started causing this and it seems to have been: https://github.com/microsoft/PowerToys/commit/1bd702340bb56ecedb198e160a91cce0ccd61b06
> 
> 
> 
> I still haven't installed the "Windows 10, version 20H2" feature update, in case that's relevant.

In current master there are some DLLs missing from setup after VCM porting. ColorPicker wasn't running in my last build.

jaimecbernardo commented:
@davidegiacometti , There's a fix on the works for that, alongside the fix for: https://github.com/microsoft/PowerToys/issues/12455
This "Settings not opening issue" seems to be related to the commit that brought in winUI 2.6

niels9001 commented:
> o be related to the comm

Hmm interesting.. does the master branch work for you? #12455 was merged earlier on.

jaimecbernardo commented:
I've built https://github.com/microsoft/PowerToys/commit/1bd702340bb56ecedb198e160a91cce0ccd61b06 and I've got this error happening there.
I've build the previous commit https://github.com/microsoft/PowerToys/commit/3c04111483042dca32d7fef20ddbbe1ed38f8cc9 and it doesn't happen there.

Now when I need to test something outside of Visual Studio I revert https://github.com/microsoft/PowerToys/commit/1bd702340bb56ecedb198e160a91cce0ccd61b06 locally in order to test it successfully.

The really annoying part here is that when I start the runner or settings from Visual Studio it just works, so I can't Debug it 🤷 .

jaimecbernardo commented:
> Hmm interesting.. does the master branch work for you? #12455 was merged earlier on.

master branch has this issue for me as well.

jaimecbernardo commented:
Hi @niels, do you know if we need the latest version?
Downgrading `Microsoft.UI.Xaml` from `2.6.1-prerelease.210709001` to `2.6.0-prerelease.210623001` seems to make settings work correctly for me again.

These symptoms (only working correctly when starting to debug) seem to indicate that `2.6.1-prerelease.210709001` is probably doing some undefined behavior in its code. I have only been able to replicate this issue in this specific machine, though.

niels9001 commented:
> Hi @niels9001 , do you know if we need the latest version?
> Downgrading `Microsoft.UI.Xaml` from `2.6.1-prerelease.210709001` to `2.6.0-prerelease.210623001` seems to make settings work correctly for me again.
> 
> These symptoms (only working correctly when starting to debug) seem to indicate that `2.6.1-prerelease.210709001` is probably doing some undefined behavior in its code. I have only been able to replicate this issue in this specific machine, though.

Hmm no idea.. I think that should be fine, as long as it's higher than 2.5 and prerelease (else it won't work with XAML Islands).

I'd suggest to file an issue against the WinUI repo though so that team can have a look at the root cause.

jaimecbernardo commented:
Opened issue here: https://github.com/microsoft/microsoft-ui-xaml/issues/5546

In the meanwhile, it might make sense to downgrade the version of Microsoft.UI.Xaml we use.

Jay-o-Way commented:
> * I think the "Welcome to PT" button/link should be white and not black in dark theme. Or is this default on WinUI 2.6 style? Why is it not the same on ligth and dark mode?

I think that's because of the AccentButton style? Don't know why a NavItem needs a style anyways?

Probably a bit off-topic for this pr, but if this button is moved to the Navigation, can we also move the "GitHub repo" and "Give feedback" links here?

> * I am not sure. But it seems for me that the space between symbol and text is to big on the "Welcome to PT" button/link.

There are a lot of manual margins all over the place. I'm really starting to question if that is good designing. I would remove most of them and just let WinUI (or XAML or whatever the language is) handle default styling. **Especially** when there are visible changes like this.

Commit:
Merge branch 'master' into users/niels9001/settings-newstyle

Commit:
Updated UI fixes and remove conflicts

Commit:
Merge branch 'master' into users/niels9001/settings-newstyle

Commit:
Adding new settings control

Commit:
PowerRename

Commit:
Sidepanel updates

Commit:
Shortcut Guide

Commit:
Updates screeens

htcfreek commented:
We should update the icons in the nav pane of the settings window to the colored ones. This would make the layout (more) consistent to oobe, process icons and the official Win11 settings app.

Jay-o-Way commented:
> update the icons on the nav pane (and FZ editor button) on the settings window to the colored ones

Goed idea. Another proof that we think alike: #12388 🙂

Commit:
General page updates

Commit:
Awake UX update

Commit:
Changed order for power preview note

Jay-o-Way commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1380,11 +1379,14 @@ From there, simply click on a Markdown file or SVG icon in the File Explorer and
     <value>Are you sure you want to delete this item?</value>
   </data>
-	<data name="Yes" xml:space="preserve">
+  <data name="Yes" xml:space="preserve">
```
Do Yes/No dialogs (or buttons) really need this? I mean, I thought it awas possible to use a `MessageBox` function with flags to determine the buttons, and they would be automatically created, localised. Or is that not possible with the languages we use?

htcfreek replied:
@Jay-o-Way
Don't forget that we don't support all languages. So using the auto translation of msgbox may lead to inconsistent ux.

niels9001 replied:
We are using a ContentDialog and those do not have a default button labels. MessageBox (for WPF) does have that I believe :)

Jay-o-Way commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -526,10 +526,10 @@ Disabling this module or closing PowerToys will unmute the microphone and camera
   </data>
   <data name="PowerRename_Toggle_EnableOnContextMenu.Header" xml:space="preserve">
-    <value>Show icon on context menu</value>
+    <value>Show PowerRename in right-click context menu</value>
```
Minor nit, but "right-click" + "context menu" sounds bit redundant?

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1376,11 +1379,17 @@ From there, simply click on a Markdown file or SVG icon in the File Explorer and
   <data name="Delete_Dialog_Description" xml:space="preserve">
     <value>Are you sure you want to delete this item?</value>
```
```suggestion
     <value>Are you sure you want to delete the item {itemToDelete}?</value>
```

@niels9001 
Can we add the name of the size to the dialog like suggested above or like this:
```
Are you sure you want to delete this item?
  <Name of size>

[yes] [no]
```

Jay-o-Way replied:
might as well strip the "..this item.." too then?

htcfreek replied:
Can you give a complete example?

Jay-o-Way replied:
"Are you sure you want to delete '{itemToDelete}'?"

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1083,7 +1086,7 @@ Made with 💗 by Microsoft and the PowerToys community.</value>
     <value>Quick layout switch</value>
   </data>
   <data name="ShortcutGuide_OpenShortcutGuide.Header" xml:space="preserve">
```
```suggestion
  <data name="ShortcutGuide_ActivateShortcutGuide.Header" xml:space="preserve">
```

Commit:
Fixes

Commit:
UI fixes

htcfreek commented:
> The hamburger menu should be removed otherwise it looks way better!

Agree. The vutton isn't important to have.

Commit:
KBM

Commit:
IsEnabled state support

Commit:
Added new controls

Commit:
Update

Commit:
Updated nugest packages

Commit:
Replaced itemscontrol

Commit:
Introducing setting button style

Commit:
FancyZones page

Commit:
Plugin page

Jay-o-Way commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -406,11 +409,11 @@ Disabling this module or closing PowerToys will unmute the microphone and camera
-  <data name="FancyZones_ExcludeApps_TextBoxControl.Header" xml:space="preserve">
-    <value>To exclude an application from snapping to zones add its name here (one per line). These apps will react only to Windows Snap.</value>
+  <data name="FancyZones_ExcludeApps.Description" xml:space="preserve">
+    <value>Excludes an application from snappting to zones and will only react to Windows Snap. Add one application name per line.</value>
```
Typo (snappting)

Jay-o-Way commented on src/settings-ui/Microsoft.PowerToys.Settings.UI.Library/ViewModels/PowerLauncherViewModel.cs:
```diff
@@ -193,113 +188,72 @@ public int MaximumNumberOfResults
-
-                    UpdateSettings();
+                    if (value == 0)
+                    {
```
They way this is written is very clear, but also extremely redundant
I would use a temporary array or something, and use the "index" as a real index.
``` cs
array = ["Dark","Light","System"]; // temporary
settings.Properties.Theme = Theme.array[value];
UpdateSettings();
```
(Same for monitor position et cetera)

niels9001 replied:
Yes, you are correct. Replaced it with a switch case.

Commit:
Switch case

Commit:
Fixed typo

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -1039,20 +1072,30 @@ Made with 💗 by Microsoft and the PowerToys community.</value>
+    <value>PowerToys Run can't provide any results without plugins</value>
+  </data>
+  <data name="Run_AllPluginsDisabled.Message" xml:space="preserve">
+    <value>Enable at least one plugin to get started</value>
```
@niels9001 
Imo we should not change text here. Please restore the old one.

```suggestion
    <value>Please enable at least one plugin</value>
```

Commit:
Plugin manager update

Commit:
Introducing SettingExpander

Jay-o-Way commented:
I'm curious to see what it would look like now :)

Commit:
Setting automation properties

Jay-o-Way commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Controls/Setting/Setting.cs:
```diff
@@ -123,9 +123,13 @@ private void Update()
+                if (_setting.ActionContent.GetType() != typeof(Button))
                 {
-                    AutomationProperties.SetName((UIElement)_setting.ActionContent, _setting.Header);
+                    // We do not want to override the default AutomationProperties.Name of a button. It's content already describes what it does.
```
It's "its" :) Sorry.

niels9001 replied:
Thanks, fixed!

Commit:
Accesibility improvements

Commit:
VCM and cleanup

Commit:
Refactoring OOBE pages part 1

htcfreek commented:
@niels9001 
I tried it in a local build. And first of all you are doing a great job. 👍🏻

I know that the work is in progress. But I want to give you a first feedback. I hope this is okay.
- [UX Controls] On ckicking the drop-down buttons they seems to hide while menu is shown. This looks wired.
- [Grouping] Some settings are grouped that have nothing to do with each other.
   - "Run at startup" and "Choose mode" on General page.
   - "Color name" is a sub item of "Default format" on ColorPicker page. (Yes they are both for picker. But one is for name and one is for color code and change one setting doesn't influence the other setting.)
   - "Opacity" is a sub item of "Choose Mode" on Shortcut Guide page. (Imo these are two different settings.)
- What about grouping the inage sizes as sub items of "add size" setting on InageResizer page like you did with exlude apps?
- I think we should combine the elements of plugin settings more.

Commit:
OOBE XAML refactoring

Commit:
Added MinWidth to actioncontent controls

Commit:
Updates to various settings

niels9001 commented:
> @niels9001

> * [UX Controls] On clicking the drop-down buttons they seems to hide while menu is shown. This looks wired.
Fixed!

> * [Grouping] Some settings are grouped that have nothing to do with each other.

Fixed!

> * What about grouping the Image sizes as sub items of "add size" setting on ImageResizer page like you did with exclude apps or Run plugins?
Not sure I can follow this one..

> * I think we should combine the elements of plugin settings more. (Maybe we should add the "expander separator rectangle" into list view style for additional plugin settings to have a separator between each additional setting.)

The expander is added to additional settings automatically - tweaked the UX a bit so that the Authored by label is a bit more 'secondary'

Thanks for the feedback!

Commit:
Spell update

Commit:
Update to author label

Commit:
Merge branch 'master' into users/niels9001/settings-newstyle

Commit:
Clean up

Commit:
Removing redunant files

Commit:
Move file

Commit:
Updated files

Commit:
Revert "Updated files"

This reverts commit 1a5c887eae4b4d4989a5e69d531d58227d73adef.

Commit:
Revert "Move file"

This reverts commit 2b06c75c1f91f7a0bbda4c61a7a732dfcedf347e.

Commit:
Revert "Removing redunant files"

This reverts commit fe79ec1701b5670c3d6298a7411964908a839dfe.

Commit:
Revert "Clean up"

This reverts commit 028e15fab6d28fa2519f5a3afa2b7e3e9c2eb297.

Commit:
Removed redundant styles

niels9001 committed:
Revert "Removed redundant styles"

This reverts commit dfdfd65021715b63bae3391518181e3ecbc81330.

niels9001 committed:
Files cleanup

niels9001 committed:
Removing converter, updating background and paddings

htcfreek commented:
@niels9001 

> > * What about grouping the Image sizes as sub items of "add size" setting on ImageResizer page like you did with exclude apps or Run plugins?

> Not sure I can follow this one..

I mean that we define the bar "add size" as expander header and move the sizes into expander content control.

> > * I think we should combine the elements of plugin settings more. (Maybe we should add the "expander separator rectangle" into list view style for additional plugin settings to have a separator between each additional setting.)
> 
> The expander is added to additional settings automatically - tweaked the UX a bit so that the Authored by label is a bit more 'secondary'

Have you updated the ux here like I suggested or did you let it as is?

Commit:
Styling updates

Commit:
Unit test updates

Commit:
Fixes

Commit:
Installer fixes

Commit:
Update Resources.resw

Commit:
Shell page updates

Commit:
Updated pagelinks

niels9001 commented:
Since this PR is near complete, I have updated the original post with all issues solved and an updated GIF with the latest state.

niels9001 commented:
> @niels9001
> I mean that we define the bar "add size" as expander header and move the sizes into expander content control.

Thought about it, but it makes keyboard navigation trickier. Also, W11 Settings does it like we have now.

>
> Have you updated the ux here like I suggested or did you let it as is?

It's very similar, but the author is at the bottom right now. Seperator lines are used for every single setting since W11 Settings does this as well.

htcfreek commented:
@niels9001 
Regarding plugin settings: we need additional a separator after each additional setting. Have you added it to the list view data template?

Commit:
Styles re-ordering

niels9001 commented:
> @niels9001
> Regarding plugin settings: we need additional a separator after each additional setting. Have you added it to the list view data template?
> 
> You can try the behavior by manipulating pt run settings json.

As you can see in the ItemTemplate, that is already included. Windows Search plugin has an additional setting:

![image](https://user-images.githubusercontent.com/9866362/130089339-226aa4c8-f55c-42da-ba4e-b1e62c3cd744.png)

Commit:
Updated converter

Commit:
Updated ToggleSwitch style

niels9001 committed:
Merge branch 'master' into users/niels9001/settings-newstyle

niels9001 commented:
This PR is now ready for review.

htcfreek commented:
@niels9001 
Sory. But I can't review it at the moment. I had an eye operation in hospital and can't see very well.

Aaron-Junker commented:
> @niels9001
> Sory. But I can't review it at the moment. I had an eye operation in hospital and can't see very well.

@htfreek Get well soon

SeraphimaZ commented:
New design looks fantastic! Really love it. Great work!
Even so, found some details to polish.

* On the File Explorer page, settings are still modifiable without elevation.

![image](https://user-images.githubusercontent.com/8949536/130101576-cd24d3d0-96e9-4ccc-bc2f-16ff446eaf8a.png)
 
 * On the Image Resizer page Filename format tooltip doesn't work.
 
![ezgif com-gif-maker](https://user-images.githubusercontent.com/8949536/130107050-8faa3de4-173d-4840-89f3-b9e9d37851db.gif)

 
* Also, after clicking `Add a size`, a new line appeared so sneaky, I didn't even notice that something had happened and thought that the button doesn't work. Don't know if it's worth fixing for now though. 

* Turning off options doesn't affect the checkboxes on the Power Rename page.
![image](https://user-images.githubusercontent.com/8949536/130106151-820a25bb-a9e6-4a36-a485-893c08fba792.png)

Commit:
Bugfixes

Commit:
Merge branch 'users/niels9001/settings-newstyle' of https://github.com/microsoft/PowerToys into users/niels9001/settings-newstyle

niels9001 commented:
@htcfreek No problem, get well soon!

@SeraphimaZ Thanks! Issues resolved in: https://github.com/microsoft/PowerToys/pull/12470/commits/628b7b97828326ef1c106801c4029dd47f0e432a

Commit:
Typo fix

jaimecbernardo committed:
[Settings] New UX - installer dll fixes (#12818)

stefansjfw commented:
![image](https://user-images.githubusercontent.com/57057282/130210340-124673ce-2a47-48ca-b073-5288e7145d79.png)
Pressing Browse button does nothing

stefansjfw commented:
![image](https://user-images.githubusercontent.com/57057282/130210542-56b76786-f5b8-498a-95a2-d062ea78562d.png)
On/Off toggle disable itself as well :) And this setting is not preserved, switching to other module settings and back in navigation pane reset it to On

stefansjfw commented:
https://github.com/microsoft/PowerToys/pull/12470
Add this one to 'Resolved issues' :)

yuyoyuppe reviewed (commented):
Outstanding work! 

There's a minor bug in autoupdating functionality on the general settings page:
![image](https://user-images.githubusercontent.com/1828123/130213507-0e342082-7fcc-4aa6-87c9-7e913f346671.png)

To test it, build the solution with version value set in `src\Version.props` to e.g. `<Version>0.42.0</Version>`

if you ever need to tweak update state manually, you could go to %localappdata%\microsoft\powertoys\updatestate.json and set "state": to 0-3

SeraphimaZ commented:
Looks like `You need to run as administator` label state is improper. It's shown while it's in administrator mode.

![image](https://user-images.githubusercontent.com/8949536/130201035-4191878f-d111-43ee-b42c-959226d661b6.png)
![image](https://user-images.githubusercontent.com/8949536/130200842-2ded9841-0e6b-4736-827b-cfe7ab42984c.png)

Jay-o-Way commented:
#9176 will be resolved with this too.

Jay-o-Way commented on .github/actions/spell-check/expect.txt:
```diff
@@ -15,6 +15,7 @@ accessibilityinsights
 Acl
 aclapi
 AColumn
+Accesible
```
this should have double S (unless it's Spanish)

niels9001 replied:
Fixed!

jaimecbernardo committed:
[Settings] New UX - Fix some failing tests (#12822)

Commit:
Bugfixes

Commit:
Typo fix

Commit:
Hiding VCM

niels9001 commented:
> ![image](https://user-images.githubusercontent.com/57057282/130210340-124673ce-2a47-48ca-b073-5288e7145d79.png)
> Pressing Browse button does nothing

@stefansjfw Saw that I was using the wrong bindings - now corrected! I'll also hide the VCM menu item for now until we have VCM in the stable branch. The Browse button didn't work on master either if I recall correctly. Maybe we can fix any VCM issues in a separate PR?

> #9176 and (I assume) #10778 will be resolved with this too.

Thanks - added those to the list. Although VCM will be hidden since it's not yet in the stable branch it has been added to and optimized with a new UX in OOBE. It's hidden in the gif in the OT.

> Looks like `You need to run as administator` label state is improper. It's shown while it's in administrator mode.

Fixed!

Commit:
Bugfixes

Commit:
Download button fix

niels9001 commented:
> Outstanding work!
> 
> There's a minor bug in autoupdating functionality on the general settings page:
> ![image](https://user-images.githubusercontent.com/1828123/130213507-0e342082-7fcc-4aa6-87c9-7e913f346671.png)
> 
> To test it, build the solution with version value set in `src\Version.props` to e.g. `<Version>0.42.0</Version>`
> 
> if you ever need to tweak update state manually, you could go to %localappdata%\microsoft\powertoys\updatestate.json and set "state": to 0-3

Should be fixed now - would you mind testing it? Unfortunately I wasn't able to test it myself with your instructions. Keeps being on "0.01" without any changes. I should run RUNNER in debug, right?

SeraphimaZ commented:
> Should be fixed now - would you mind testing it? Unfortunately I wasn't able to test it myself with your instructions. Keeps being on "0.01" without any changes. I should run RUNNER in debug, right?

Tested, also with other states, it fixed.

![image](https://user-images.githubusercontent.com/8949536/130243721-00d4e3d7-4997-4151-9a51-93c2c8b56cc3.png)
![image](https://user-images.githubusercontent.com/8949536/130243800-a92cb8ac-bc1b-41f5-8820-ddf078f2574a.png)
![image](https://user-images.githubusercontent.com/8949536/130243914-a2cf4156-152e-44d9-9c94-2cd3a3775d9a.png)

SeraphimaZ commented:
> Unfortunately I wasn't able to test it myself with your instructions. Keeps being on "0.01" without any changes. I should run RUNNER in debug, right?

Rebuild is needed here, version won't be changed otherwise.

htcfreek commented:
> > Should be fixed now - would you mind testing it? Unfortunately I wasn't able to test it myself with your instructions. Keeps being on "0.01" without any changes. I should run RUNNER in debug, right?
> 
> Tested, also with other states, it fixed.
> 
> ![image](https://user-images.githubusercontent.com/8949536/130243721-00d4e3d7-4997-4151-9a51-93c2c8b56cc3.png)
> ![image](https://user-images.githubusercontent.com/8949536/130243800-a92cb8ac-bc1b-41f5-8820-ddf078f2574a.png)
> ![image](https://user-images.githubusercontent.com/8949536/130243914-a2cf4156-152e-44d9-9c94-2cd3a3775d9a.png)
> 

As I mentioned in the related issue we should chose the yellow attention state for "update is available" message. Because this is something the user should pay attention and not something that is good as is.

htcfreek commented:
@niels9001 , @SeraphimaZ 
Can you please try how it looks if we have more than one additional setting per plugin. On my last test (last week) the splitter lines are missing between the settings.

You can simulate it by adding an additional settings entry to PT Run's settings.json. (Attention: The entry should be added manually after PT Run is initialized/started. While starting up the json is updated by PT Run.)

Jay-o-Way commented:
Silghtly off-topic request: is it possible to make a video, instead of a gif, for extended visual previews like in the first comment? Can't pause or scroll through a gif :(

htcfreek commented:
@niels9001 , @SeraphimaZ 
What about adding a "Release notes" link behind the version on general page?

![130243721-00d4e3d7-4997-4151-9a51-93c2c8b56cc3.png](https://user-images.githubusercontent.com/61519853/130291707-830d4229-c938-4ec7-8d10-79515ac13d7b.png)

Commit:
Remove Newtonsoft.Json from installer file

niels9001 commented:
@htcfreek 
> Can you please try how it looks if we have more than one additional setting per plugin. On my last test (last week) the splitter lines are missing between the settings.

Here you go :)

![image](https://user-images.githubusercontent.com/9866362/130317063-eedaefc3-ba02-4c0a-9f14-1c7ecad75e61.png)


I have made some adjustments to the update bars as well: the "what's new" link is moved to the right, and the action button near where the progress loader will be shown after clicking it. The download buttons will disappear when the loader is active - @SeraphimaZ would you mind doing a last sanity check to see if all the logic works :)?

@htcfreek Release notes link has been added + link to release notes.
![image](https://user-images.githubusercontent.com/9866362/130322776-68f02114-417d-4596-8e7d-ac49700fe141.png)

Commit:
Updated visuals

Commit:
Bugfixes

htcfreek commented:
@niels9001 
1. Great. Seems that the bug of one bate for all additional settings is fixed.
2. The Update ux looks better now.
3. Do we have an informative message (bar) on dev build that updater isn't available?

htcfreek commented:
@niels9001 
4. Have you changed the "update ready to install" message into warning layout too?

htcfreek commented:
@niels9001 
Have you payed attention to the changes of #12560 (PT plugin list search)?

Jay-o-Way commented:
I disagree on the ⚠️ icon for an update. ℹ️ should be good here.

htcfreek commented:
> I disagree on the ⚠️ icon for an update. ℹ️ should be good here.

Imo it only should not be a green "ready" checkmark. Because a green checkmark indicates something is done/okay.

Why do you think we should use a ℹ️ symbol instead of a ⚠️ symbol? I like to understand your point.

Jay-o-Way commented:
Because an update - to me - is not that important. I feel warnings are appropriate when there is danger ahead. Either something is failing or there is a situation where data can be lost, something like that.
I mean, maybe when you account for the _current_ status - that is: heavily under construction - you might have a point. But, over time, it becomes more developed and stable. I'm the kind of guy that will check for updates when it suits *me*, and being told "I have to" will mostly achieve a very different result. I literally think "hey, there's this update, but I'll leave it to you what to do with it" is a polite and respectful manner.

Jay-o-Way commented:
In other words:
## Severity
Avoid setting the Severity property for a notification that does not match the information communicated in the Title, Message, or custom content.

The accompanying information should aim to communicate the following to use that Severity.

- Error: An error or problem that has occurred.
- Warning: A condition that might cause a problem in the future.
- Success: A long-running and/or background task has completed.
- Default: General information that requires the user's attention.

Icons and color should not be the only UI components signifying meaning for your notification. Text in the notification's Title and/or Message should be included to display information.

niels9001 commented:
@htcfreek Yes, search is already integrated.

@Jay-o-Way @htcfreek Valid points. Since we have other warning messages on the page already, I'll just leave it as is to make it more stand-out. Ideally the 'information' severity level would be ideal, but it doesn't really stand out (flagged that here: https://github.com/microsoft/microsoft-ui-xaml/issues/5741)

Let's park this discussion (and other minor points) for now or create a separate issue. It's a big PR to merge and I'd like to focus on making sure we don't have any bugs or regressions showing up.

htcfreek commented:
> @htcfreek Yes, search is already integrated.
> 
> @Jay-o-Way @htcfreek Valid points. Since we have other warning messages on the page already, I'll just leave it as is to make it more stand-out. Ideally the 'information' severity level would be ideal, but it doesn't really stand out (flagged that here: https://github.com/microsoft/microsoft-ui-xaml/issues/5741)
> 
> Let's park this discussion (and other minor points) for now or create a separate issue. It's a big PR to merge and I'd like to focus on making sure we don't have any bugs or regressions showing up.

@niels9001 
Okay. In this case I agree to you.
Then let's do only point 2 (move "whats new" link) and skip the points 3 and 4.

Commit:
Syntax fix

Commit:
Updated installer file

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Views/GeneralPage.xaml:
```diff
@@ -7,227 +7,232 @@
+                    <!-- Ready to install -->
+                    <muxc:InfoBar x:Uid="General_NewVersionReadyToInstall"
+                                  IsClosable="False"
+                                  Severity="Success"
```
```suggestion
                                  Severity="Warning"
```

htcfreek commented on src/settings-ui/Microsoft.PowerToys.Settings.UI/Views/GeneralPage.xaml:
```diff
@@ -7,227 +7,232 @@
+                    <!-- New version available -->
+                    <muxc:InfoBar x:Uid="General_NewVersionAvailable"
+                                  IsClosable="False"
+                                  Severity="Success"
```
```suggestion
                                  Severity="Warning"
```

htcfreek replied:
@niels9001 
Is it a mistake marking this as solved while keeping "success"?

(For me an available update isn't something that is success.)

htcfreek reviewed (changes requested):
I agree with @niels9001 that it makes no sense to use a severity style which is difficult to see on some color modes. But I think this updating states are definitely not of the type "success". In addition to the points before "Windows Update" uses the warning state on available updates too. So I think we should switch both to the severity type warning. 

In addition we can open an issue to update style to information when the WinUI colors are improved.

htcfreek commented:
# !!!! BUG BASH !!!!

@niels9001 
I have found some bugs and have some small improvements. Here they are grouped by app/window.

### Settings window
1. When hovering the "restart as admin" button the expander arrow is hovered too. Is this normal behavior? **IMAGE coming soon**
2. The drop downs of drop down buttons only respect "Windows default color" regardless of the setting in PT settings.
3. The tool tips only respect "Windows default color" regardless of the setting in PT settings.
4. The delete an image size on IR page only respect "Windows default color" regardless of the setting in PT settings.
5. When disabling modules or settings the radio buttons have a black dot on it. I would expect that the dot is grey or something else then. (Only on dark mode.) **IMAGE coming soon**
6. For me the plugin icons (except of Unit converter) are missing on PT Run settings page.
7. When I disabling the PowerRename module the settings (except "Maximum number of items") aren't disabled.
8. Win11 has the list search fields between header/description and the list. What about moving the plugin list search filed from the side to under the description text? **IMAGE coming soon**

### KBM UX
1. The KBM UX for adding new remapping doesn't has WinUI 2.6 style. Do we have an issue here?

### OOBE pages
1. The "Open settings" button has no settings icon anymore. Is this a bug or a layout improvement?

htcfreek commented:
@niels9001 

> Then let's do only point 2 (move "whats new" link) ...

Have you done this?

niels9001 commented:
> # !!!! BUG BASH !!!!
> @niels9001
> I have found some bugs and have some small improvements. Here they are grouped by app/window.
> 
> ### Settings window
> 1. When hovering the "restart as admin" button the expander arrow is hovered too. Is this normal behavior? ![image]
Thanks, fixed.

> 2. The drop down lists of drop down buttons only respect "Windows default color" regardless of the setting in PT settings.
> 3. The tool tips only respect "Windows default color" regardless of the setting in PT settings.
> 4. The delete an image size on IR page only respect "Windows default color" regardless of the setting in PT settings.
> 5. When disabling modules or settings the radio buttons have a black dot on it. I would expect that the dot is grey or something else then. (Only on dark mode.)
This is all part of WinUI - there are some known issues with WinUI 2.6 and XAML Islands. The Acrylic doesn't render correctly. Will be resolved with later versions of WinUI.

> 6. For me the plugin icons (except of Unit converter) are missing on PT Run settings page.
For me a Unit Converter and Settings aren't rendering. But no logic changes there, so I guess the same issue is on the master branch version of Settings?

> 7. When I disabling the PowerRename module the settings (except "Maximum number of items") aren't disabled.
Thanks, fixed.
> 8. Win11 has the list search fields between header/description and the list. What about moving the plugin list search filed from the side to under the description text?
Feels cramped and not a real benefit. W11 Settings actually puts this above the card itself. But that looks weird - will leave it for now.
 
> ### KBM UX
> 1. The KBM UX for adding new remapping doesn't has WinUI 2.6 style. Do we have an issue here?
This is known. KBM is not part of settings, so out of scope.
 
> ### OOBE pages
> 1. The "Open settings" button has no settings icon anymore. Is this a bug or a layout improvement?
Removed for ease of dev.

Commit:
Bugfixes

Jay-o-Way commented:
KBM pop-up is #6664

htcfreek commented:
@niels9001 
If you answer on md quotes it would be nice when you add an empty line between quote and answer. Otherwise your answer is shown as part of the quote on the GitHub app.

**Better:**
```markdown
> Quote text.

Answer text.
```

htcfreek commented:
> > For me the plugin icons (except of Unit converter) are missing on PT Run settings page.

> For me a Unit Converter and Settings aren't rendering. But no logic changes there, so I guess the same issue is on the master branch version of Settings?

I have shutdown my computer. But it might help to delete PT Run's settings.json file. Does it?

justingolden21 commented:
I'm just happy to see the scrollwheel seems to work in this version lol

niels9001 commented:
> I'm just happy to see the scrollwheel seems to work in this version lol

It wasn't before for you? I know there's this issue: https://github.com/microsoft/PowerToys/issues/3400

Can you verify that on the current stable build scrolling with a scroll-wheel doesn't work and it works for this branch?

justingolden21 commented:
> > I'm just happy to see the scrollwheel seems to work in this version lol
> 
> It wasn't before for you? I know there's this issue: #3400
> 
> Can you verify that on the current stable build scrolling with a scroll-wheel doesn't work and it works for this branch?

Thanks for the quick reply. It doesn't work on the current version installed on my PC. I'm not sure if it does in this branch or not, I was just commenting on the video shown at the top of the thread where the user seems to be scrolling. I can't seem to scroll in my current app (among other bugs) and I can scroll perfectly fine in all other apps, including Microsoft ones and default windows apps. In addition to the scroll wheel not working, the scroll bar is buggy (often doesn't work, hard to select, and sometimes teleports) and other buttons such as arrow keys, page up and page down, home and end do not work either in any part of power tools.

SeraphimaZ commented:
> @SeraphimaZ would you mind doing a last sanity check to see if all the logic works :)?

Checked, everything works well except one small glitch with an inactive toggle color on the general page. 

Steps to repro:
1. Set dark theme
2. Close settings.
3. Open settings.
4. Choose a light theme.
Result:

![image](https://user-images.githubusercontent.com/8949536/130424575-4cee501c-2c0d-442e-9a90-d2fb30c412ab.png)

It doesn't happen, when you just change the theme from light to dark and back to light.

htcfreek commented:
> > I'm just happy to see the scrollwheel seems to work in this version lol
> 
> It wasn't before for you? I know there's this issue: https://github.com/microsoft/PowerToys/issues/3400
> 
> Can you verify that on the current stable build scrolling with a scroll-wheel doesn't work and it works for this branch?

What I know is that scrolling with `Page up` and `Page down` doesn't work on master branch. I didn't try scroll wheel in the past.

Commit:
New label

niels9001 commented:
@htcfreek @Jay-o-Way The flyout/tooltip issue with acrylic seems to be resolved in WinUI 2.7.x. So upgrading to that will resolve that issue as can be seen here: https://github.com/microsoft/microsoft-ui-xaml/issues/5277

jaimecbernardo commented:
@niels9001 , upgrading to WinUI to 2.7.x will break the settings on Windows versions we still support.
I've just tested this: https://github.com/microsoft/microsoft-ui-xaml/issues/5435#issuecomment-903635572

niels9001 commented:
> @niels9001 , upgrading to WinUI to 2.7.x will break the settings on Windows versions we still support.
> I've just tested this: [microsoft/microsoft-ui-xaml#5435 (comment)](https://github.com/microsoft/microsoft-ui-xaml/issues/5435#issuecomment-903635572)

Thanks for checking! Seems that we'll have to upgrade later (once the crashing bug is fixed) to resolve it. I created a tracking issue for that here: #12846

Commit:
Theming support for OOBE

htcfreek commented:
@niels9001 
The docs link of #12598 is missing in the new settings UX.

Commit:
Put back in Documentation hyperlink

niels9001 commented:
> @niels9001
> The docs link of #12598 is missing in the new settings UX.

Thanks for flagging, must of missed that when resolving conflicts. Put it back in!

![image](https://user-images.githubusercontent.com/9866362/130471076-97004852-6eaa-44e4-a0e5-345eff9aee4d.png)

Commit:
Right uid

htcfreek commented:
> > > For me the plugin icons (except of Unit converter) are missing on PT Run settings page.
> 
> > For me a Unit Converter and Settings aren't rendering. But no logic changes there, so I guess the same issue is on the master branch version of Settings?
> 
> I have shutdown my computer. But it might help to delete PT Run's settings.json file. Does it?
> 
> Maybe I can check tomorrow on master branch. If the issue exist there I will open an issue. Imo this should be fixed until next release.

Deleting PT Run's `settings,json` fixes the bug. All icons are shown.

htcfreek commented:
@niels9001 
What about adding a restart button into the "Please restart as admin" warning bar (except on general page)? (I can open a new issue if you want me to do so.)

htcfreek commented:
@niels9001 
- I think a shortcut symbol makes more sense here than an window symbol.

![image](https://user-images.githubusercontent.com/61519853/130475294-a5bc5136-28b5-42fe-a3f0-9993d5b497d3.png)

niels9001 commented:
> @niels9001
> 
> 1. What about adding a restart button into the "Please restart as admin" warning bar (except on general page)? (I can open a new issue if you want me to do so.)

Yes please open a new issue for this.

> 2. Can we use expanded style here. For me it looks bad to have title and note in one column and I can say it's not eye friendly for people with disabilities. And it is more friendly to write "Please enable ..."!

Default InfoBar layout - not going to change that in this PR. "Please enable" is not used across Windows.

>    ![image](https://user-images.githubusercontent.com/61519853/130476128-e843695b-d9b2-4e24-a982-82afb0cd1377.png)
> 3. I think a shortcut symbol makes more sense here than an window symbol. ![image](https://user-images.githubusercontent.com/61519853/130475294-a5bc5136-28b5-42fe-a3f0-9993d5b497d3.png)

This is about Layout related settings (not only quick layouts, that relates to shortcuts). We already use shortcut for the action key setting.

> 4. It makes more sense to have this message after "Global results" checkbox. ![image](https://user-images.githubusercontent.com/61519853/130476446-67d51675-2aec-4fbd-8b2e-9b2792764145.png)

But it's related to the keyword text field? Makes sense to have the warning as close as possible to that IMO.

niels9001 commented:
@jaimecbernardo OK if we merge this?

Aaron-Junker commented:
@niels9001 Will you change the labels of the corresponding issues or should I do it. I have currently some time, so it would not be a problem if you want

htcfreek commented:
> > @niels9001
> > 
> > 1. What about adding a restart button into the "Please restart as admin" warning bar (except on general page)? (I can open a new issue if you want me to do so.)
> 
> Yes please open a new issue for this.

I will do that later.

Can you please open an issue for changing "succes" type of update available/ready messages to "info" if winui bug is solved.

> > 1. Can we use expanded style here. For me it looks bad to have title and note in one column and I can say it's not eye friendly for people with disabilities. And it is more friendly to write "Please enable ..."!
> 
> Default InfoBar layout - not going to change that in this PR. "Please enable" is not used across Windows.
> 

okay.

> > 1. I think a shortcut symbol makes more sense here than an window symbol.
> 
> This is about Layout related settings (not only quick layouts, that relates to shortcuts). We already use shortcut for the action key setting.
> 

okay

> > 1. It makes more sense to have this message after "Global results" checkbox.
> 
> But it's related to the keyword text field? Makes sense to have the warning as close as possible to that IMO.

But in the message we write "**Action key** or **global results**". So I thought it makes more sense to put it after both settings instead of between both settings.

Jay-o-Way commented:
By the way, will `/doc/images/icons/FancyZones_MDL2.svg` be deleted with this pr?

niels9001 commented:
> @niels9001 Will you change the labels of the corresponding issues or should I do it. I have currently some time, so it would not be a problem if you want

Feel free to do so ;)

jaimecbernardo commented:
> @jaimecbernardo OK if we merge this?

There have been enough eyes here. If all feedback has been addressed, it can be squash and merged.
Great work!

htcfreek commented:
> > > 1. It makes more sense to have this message after "Global results" checkbox.
> > 
> > 
> > But it's related to the keyword text field? Makes sense to have the warning as close as possible to that IMO.
> 
> But in the message we write "**Action key** or **global results**". So I thought it makes more sense to put it after both settings instead of between both settings. Maybe it makes sense to put above "activation command".

@niels9001 
Because of a second point I will open a new issue here tomorrow.

Aaron-Junker commented:
> > @niels9001 Will you change the labels of the corresponding issues or should I do it. I have currently some time, so it would not be a problem if you want
> 
> Feel free to do so ;)

@niels9001 Done.

Aaron-Junker commented:
Big shout outs to all developers who worked on this PR and tested it. (And of course also all other developers who work on this amazing project.) I wasn't able to test it by myself, but it looks amazing. We're one of the first not core OS apps which adopted the new Windows 11 design. From my perspective this is amazing. Keep the good work up!

htcfreek commented:
@niels9001 
One last question I missed to ask: 
Why do I have to rebuild the complete PT project to get changes take affect? If I only rebuild settings project the build result is not in the correct directory.

niels9001 commented:
> @niels9001 
> One last question I missed to ask: 
> Why do I have to rebuild the complete PT project to get changes take affect? If I only rebuild settings project the build result is not in the correct directory.

You need to run PowerToys.Settings.. but that is only the window. The entire UI (= XAML Island) is PowerToys.Settings.UI. so if you build only .UI you can just run PowerToys.Settings.
